File: libiqtree2/tests/python/test_calculate_RF_distance.py
import sys
import os
sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..','libs')))
import libiqtree2 as iqtree  # Adjust the import according to your actual module name

def test_calculate_RF_distance():
    tree1 = "(A,B);"
    tree2 = "(B,A);"
    assert iqtree.calculate_RF_distance(tree1, tree2) == 0  # Assuming 0 is the expected distance

File: libiqtree2/tests/c++/src/test_calculate_RF_distance.cpp
// File: test_calculate_RF_distance.cpp

#ifdef CATCH2_OLD_HEADER
    #include <catch2/catch.hpp>
#else
    #include <catch2/catch_all.hpp>
#endif
#include "libiqtree2_functions.h"
#include <iostream>

TEST_CASE("calculate_RF_distance returns expected distance", "[calculate_RF_distance]") {
    std::string tree1 = "(A,B);";
    std::string tree2 = "(B,A);";
    int rfdist = calculate_RF_distance(tree1, tree2);
    std::cout << "rfdist = " << rfdist << std::endl;
    REQUIRE(calculate_RF_distance(tree1, tree2) == 0); // Assuming 0 is the expected distance for identical/similar trees
}
